drivers/soundwire/intel.c

```diff
@@ -881,19 +881,36 @@ static int
 intel_hw_free(struct snd_pcm_substream *substream, struct snd_soc_dai *dai)
 {
 	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
+	struct sdw_intel *sdw = cdns_to_intel(cdns);
 	struct sdw_cdns_dma_data *dma;
 	int ret;
 
 	dma = snd_soc_dai_get_dma_data(dai, substream);
 	if (!dma)
 		return -EIO;
 
+	ret = sdw_deprepare_stream(dma->stream);
```

Comment:
Not introduced in this patch but "deprepare" as a verb sounds very odd. I note there are also separate "release" and "free" methods... quite confusing.

Reply:
it's what the API is, I can't change it now.
Deprepare actually makes sense, it's tied to the prepare/de-prepare step in SoundWire.
release is the mirror of alloc. but yes why release instead of free I don't know.

Reply:
If it's in SoundWire specs as well, then makes sense. Anyway, cannot change in this series. Ok for me.